test/unit/tuple_test/device/tuple.hpp

```diff
@@ -30,113 +30,32 @@
 #include <rocwmma/internal/io_traits.hpp>
 #include <rocwmma/internal/layout.hpp>
 #include <rocwmma/internal/tuple.hpp>
+#include <rocwmma/internal/vector.hpp>
 #include <rocwmma/rocwmma.hpp>
 
 static constexpr uint32_t ERROR_VALUE   = 7u;
 static constexpr uint32_t SUCCESS_VALUE = 0u;
 
 namespace rocwmma
 {
-    __device__ static inline bool isTupleTest()
-    {
-        bool err = false;
-
-        err |= std::is_tuple<int>::value;
-        err |= !std::is_tuple<std::tuple<int, float>>::value;
-        err |= std::is_tuple<const std::tuple<int, float>>::value;
-        err |= std::is_tuple<std::tuple<int, float>&>::value;
-
-        return err;
-    }
-
     __device__ static inline bool operatorMultTest()
     {
         bool err = false;
 
-        auto srcTuple          = std::make_tuple(1, 2.0, 3u);
-        auto expectHeadElement = std::make_tuple(2);
-        auto resultHeadElement = std::operator_mult_impl(2, srcTuple, std::index_sequence<0>{});
-        err |= expectHeadElement != resultHeadElement;
+        auto const srcTuple          = make_vector(1, 2, 3);
+        auto       expectHeadElement = make_vector(2);
+        auto       resultHeadElement = detail::mult_poly_vec_impl(
+            make_vector(2, 2, 2), srcTuple, detail::index_sequence<0>{});
+        err |= vector_reduce_and(expectHeadElement != resultHeadElement);
```

Comment:
Do we need to add `vector_reduce_and` around a Boolean result?

Reply:
Vector operations have a boolean vector result. These vectors need to be reduced